### PR TITLE
tests/SearchBar and other

### LINKED ISSUE #965: Search Bar Unit Tests

### PR DESCRIPTION
**Pull request by sirugh**
These be tests for the `SearchBar` component directory. 

Closes #965

### CODE REVIEW

**Comment by awilcoxa** [comment]
needs an issue

**Comment by sirugh** [comment]
@jimbo This is ready to go but I rebased onto release/3.0 and now I can't get it to go back to 2.0 cleanly.

**Comment by sirugh** [comment]
@tjwiebell I brought the `SearchBar` tests to 100% but I'm going to opt out of doing index or container as they are indeed boilerplate.

File: packages/venia-concept/src/components/SearchBar/__tests__/searchBar.spec.js [hunk 1]
```diff
@@ -182,3 +207,20 @@ test('handles clicks inside and out of autocomplete input', () => {
...
+    instance.children[0].instance.componentWillUnmount();
+
+    expect(removeEventListenerMock).toHaveBeenCalledWith(
+        'mousedown',
+        instance.children[0].instance.autocompleteClick,
+        false
+    );
+});
+
+test;
```
Comment (tjwiebell) [inline packages/venia-concept/src/components/SearchBar/__tests__/searchBar.spec.js]: :baby:

Comment (sirugh) [reply packages/venia-concept/src/components/SearchBar/__tests__/searchBar.spec.js]: uhhhh. https://www.youtube.com/watch?v=Qv5fqunQ_4I

**Pull request merged**